### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by cvalencia1991**
# In this milestone, I did the following activities:

- Add Redux Files
- Add books and categories in the Redux File 
- Configure books and categories in the dock pather
- Configure reduces in the configure store 
- Fix linters Errors

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const ADD = 'ADD';
+const REMOVE = 'REMOVE';
```
Comment (codecaiine) [inline src/redux/books/books.js]: - [ ] Good job so far From but from [ducks pattern](https://github.com/erikras/ducks-modular-redux)...` const CREATE = 'my-app/widgets/CREATE';` so you should have something like this `const ADD_BOOK = bookstore/books/ADD_BOOK ` and  `const REMOVE_BOOK = bookstore/books/REMOVE_BOOK`. Please fix it to make your code works as expected :smile: 

#### Find an example below :arrow_down: 

```
const ADD_BOOK = bookstore/books/ADD_BOOK ;
const REM_BOOK = bookstore/books/REM_BOOK;

// Reducer Section
export default function reducer(state = [ ], action = {}) {
  switch (action.type) {
    case ADD_BOOK:
      return [
        ...state,
        action.payload,
      ];
    case REM_BOOK:
      return [
        state.filter((item) => item !== action.payload),
      ];
    default:
      return state;
  }
}

// Action Creators Section
export function addBook(obj) {
  return { type: ADD_BOOK, payload: obj };
}

export function removeBook(obj) {
  return { type: ADD_BOOK, payload: obj };
}

```

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const CHECKING = 'CHECK';
```
Comment (codecaiine) [inline src/redux/categories/categories.js]: - [ ] Good job so far but from [ducks pattern](https://github.com/erikras/ducks-modular-redux)...` const CREATE = 'my-app/widgets/CREATE';` so you should have something like this `const CHECK_STATUS = 'bookstore/categories/CHECK_STATUS'; `. Please fix it :+1: 

#### Find an example below :arrow_down: 

```

// Actions
const CHECK_STATUS = 'bookstore/categories/CHECK_STATUS';

// Reducer
export default function categoriesReducer(state = [], action) {
  switch (action.type) {
    case CHECK_STATUS:
      return 'Under Construction';
    default:
      return state;
  }
}

// Action creators
export function checkStatus() {
  return {
    type: CHECK_STATUS,
  };
}

```

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @cvalencia1991 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Descriptive PR :heavy_check_mark: 
- No linters errors :heavy_check_mark: 
- Good project setup :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @cvalencia1991 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Descriptive PR :heavy_check_mark: 
- No linters errors :heavy_check_mark: 
- Good project setup :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categories.js (hunk 2 above)
Comment (cvalencia1991) [reply src/redux/categories/categories.js]: ok i change the way you told me that i need to put it

File: src/redux/books/books.js (hunk 1 above)
Comment (cvalencia1991) [reply src/redux/books/books.js]: ok i change the duck pattern that you told me to put it

**Review by Reem-lab — APPROVED** [review]
# Hi @cvalencia1991,👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉🔥🔥💯💯



## Highlights ✨✨
📌 Followed Gitflow ✅
📌 Great working on the functionality ✅
📌No Linters error ✅
📌Descriptive Pull request ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by cvalencia1991** [comment]
thank you very much!!!

**Pull request merged**